src/model.ts

```diff
@@ -686,16 +677,14 @@ export function build<TSchema extends BaseSchema, TDefaults extends Partial<TSch
         {
           ...model.defaultOptions,
           ...options,
-        } as CollectionInsertOneOptions
+        } as InsertOneOptions
       );
 
-      if (
-        result.result.ok === 1 &&
-        result.result.n === docs.length &&
-        result.insertedCount === docs.length &&
-        result.ops.length === docs.length
-      ) {
-        return result.ops as TSchema[];
+      if (result.acknowledged && result.insertedCount === docs.length) {
+        return documents.map((doc, index) => ({
```

Comment:
I am not sure if we want to start deviating from the driver, it kind of goes against the philosophy behind this library.